components/builder-api/src/server/handlers.rs

```diff
@@ -641,3 +641,78 @@ fn github_creds_from_body<S: Into<String>>(
     }
     Ok(token)
 }
+
+pub fn create_project_integration(req: &mut Request) -> IronResult<Response> {
+    let params = match validate_params(req, &["origin", "name", "integration"]) {
+        Ok(p) => p,
+        Err(st) => return Ok(Response::with(st)),
+    };
+
+    let body = req.get::<bodyparser::Json>();
+    match body {
+        Ok(Some(_)) => (),
+        Ok(None) => {
+            warn!("create_project_integration: Empty body in request");
+            return Ok(Response::with(status::BadRequest));
+        }
+        Err(e) => {
+            warn!("create_project_integration, Error parsing body: {:?}", e);
+            return Ok(Response::with(status::BadRequest));
+        }
+    };
+
+    // We know body exists and is valid, non-empty JSON, so we can unwrap safely
+    let json_body = req.get::<bodyparser::Raw>().unwrap().unwrap();
+
+    let mut opi = OriginProjectIntegration::new();
+    opi.set_origin(params["origin"].clone());
+    opi.set_name(params["name"].clone());
+    opi.set_integration(params["integration"].clone());
+    opi.set_integration_name(String::from("default"));
+    opi.set_body(json_body);
+
+    let mut request = OriginProjectIntegrationCreate::new();
+    request.set_integration(opi);
+
+    match route_message::<OriginProjectIntegrationCreate, NetOk>(req, &request) {
+        Ok(_) => Ok(Response::with(status::NoContent)),
+        Err(err) => {
+            if err.get_code() == ErrCode::ENTITY_CONFLICT {
+                warn!("Failed to create integration as it already exists");
+                Ok(Response::with(status::Conflict))
+            } else {
+                error!("create_project_integration:1, err={:?}", err);
+                Ok(Response::with(status::InternalServerError))
+            }
+        }
+    }
+}
+
+pub fn get_project_integration(req: &mut Request) -> IronResult<Response> {
+    let params = match validate_params(req, &["origin", "name", "integration"]) {
+        Ok(p) => p,
+        Err(st) => return Ok(Response::with(st)),
+    };
+
+    let mut opi = OriginProjectIntegration::new();
+    opi.set_origin(params["origin"].clone());
+    opi.set_name(params["name"].clone());
+    opi.set_integration(params["integration"].clone());
+    opi.set_integration_name(String::from("default"));
```

Comment:
Are there plans for there to be other integration names in the future?

Might be worth making a constant for this.

Reply:
Yes, eventually we will have named integrations. Will add a constant.